pyccel/parser/syntactic.py

```diff
@@ -142,6 +147,9 @@ def __init__(self, inputs, **kwargs):
         self._code    = code
         self._context = []
 
+        # provides information about the calling context to collect constants and functions
+        self._context_dict = context_dict or {}
```

Comment:
Do you accept `None` as an argument? Maybe we should set `None` as the default value

Reply:
Fixed [7d7ceb8](https://github.com/pyccel/pyccel/pull/2382/commits/7d7ceb8e22abc9b80796dba998899c3ad5d3a7e0)